src/modules/app/app.ts

```diff
@@ -45,20 +47,38 @@ export class App {
 
     public start(): void {
         this.cart.drawCart();
-        this.cart.setCurrentValues('0', '0');
+        this.cart.setCurrentValues(this.getSum(), this.getCount());
         this.cart.resetBtn(); // will be remove at the end
-        this.globalFiltres.createFilters(productsData);
 
-        const startFilters: Map<FilterType, Set<string>> = this.globalFiltres.getCurrentFilters();
-        const startSliders: Map<SliderType, SliderValue> = this.globalFiltres.getCurrentSliders();
-        this.updateResult(startFilters, startSliders, '');
+        const searchParams = Object.fromEntries(new URLSearchParams(window.location.search));
 
-        const btnCopy = new Button(
-            document.querySelector('.buttons__wrapper')!,
-            (type: string) => this.onClickButton(type)
-        );
-        btnCopy.drawButton('copy');
-        btnCopy.drawButton('reset');
+        if (Object.keys(searchParams).includes('id')) {
+            const productView = new ProductPage(document.querySelector('.main__container')!, Number(searchParams.id));
+            productView.drawProductPage();
+        } else {
+            this.globalFiltres.createFilters(productsData);
+            if (Object.keys(searchParams).length !== 0) {
+                for(const key in searchParams) {
+                    if (key === 'price' || key === 'stock') {
+                        const objectMinMax: SliderValue = {min: Number(searchParams[key]!.split('/')[0]), max: Number(searchParams[key]!.split('/')[1])};
+                        this.globalFiltres.setCurrentSliders(key as SliderType, objectMinMax);
+                    } else if (key === 'category' || key === 'brand') {
+                        searchParams[key]!.split('|').forEach((item) => {
+                            const checkbox =  document.getElementById(item!)! as HTMLInputElement;
+                            checkbox.checked = true;
+                            this.globalFiltres.setCurrentFilters(key as FilterType, item);
+                        })
+                    }
+                }
+            }
```

Comment:
It's ok for now if it works. In real life I would suggest move this to function and cover with tests. So much going on here..